gatsby-plugin-material-ui/src/gatsby-node.js

```diff
@@ -0,0 +1,73 @@
+import fs from 'fs';
+import path from 'path';
+import os from 'os';
+
+// Write out the gatsby-mui-config module to .cache.
+// Happens before webpack starts compiling page.
+// .cache is available when gatsby-browser.js and gatsby-ssr.js runs.
+
+// eslint-disable-next-line import/prefer-default-export
+export const onPreBootstrap = ({ store }, pluginOptions) => {
+  const { program } = store.getState();
+
+  const fixPath = module => {
+    let newModule = module;
+    if (os.platform() === `win32`) {
+      newModule = module.split(`\\`).join(`\\\\`);
+    }
+    return newModule;
+  };
+
+  let module;
+
+  // Not allowed to specify both pathToStylesProvider and StylesProvider props
+  // in gatsby-config.js
+  try {
+    if (pluginOptions.pathToStylesProvider && pluginOptions.stylesProvider) {
+      throw new Error(
+        `You specified both pathToStylesProvider and stylesProvider in gatsby-config.js. Only one should be specified...`,
+      );
+    }
+
+    if (pluginOptions.stylesProvider) {
+      console.log(
+        `pluginOptions.stylesProvider:`,
+        pluginOptions.stylesProvider,
+      );
+      module = `const stylesProviderProps = ${JSON.stringify(
+        pluginOptions.stylesProvider,
+      )};
+
+export default stylesProviderProps;
+`;
+      console.log(`module:`, module);
```

Comment:
What is this console output for? For debug messages please use the npm package debug. For other messages, see https://www.gatsbyjs.org/docs/node-api-helpers/#reporter